src/renderer/components/general-settings/general-settings.js

```diff
@@ -104,8 +104,10 @@ export default Vue.extend({
 
       Object.keys(this.$i18n.messages).forEach((locale) => {
         const localeName = this.$i18n.messages[locale]['Locale Name']
-        if (typeof localeName !== 'undefined') {
+        if (typeof localeName !== 'undefined' && locale !== 'system') {
           names.push(localeName)
+        } else if (typeof localeName !== 'undefined' && locale === 'system') {
+          names.push('Follow System')
         } else {
           names.push(locale)
         }
```

Comment:
A better structure for this would be:
```js
if (locale === 'system') {
  names.push('Follow System')
  return
}

const localeName = this.$i18n.messages[locale]['Locale Name']
if (typeof localeName !== 'undefined') {
  names.push(localeName)
} else {
  names.push(locale)
}

Reply:
Thanks for the review. I wasn't aware of the `return` statement skipping the loop-run :see_no_evil:. 
I will clean that mess up